File: Dot_Net_Project/src/DevOpsShack.ProjectOps/Dtos/ProjectRequest.cs
using System.Text.Json.Serialization;

namespace DevOpsShack.ProjectOps.Dtos;

public class ProjectRequest
{
    [JsonPropertyName("title")]
    public string? Title { get; set; }

    [JsonPropertyName("category")]
    public string? Category { get; set; }

    [JsonPropertyName("difficulty")]
    public string? Difficulty { get; set; }

    [JsonPropertyName("status")]
    public string? Status { get; set; }

    [JsonPropertyName("owner")]
    public string? Owner { get; set; }

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    [JsonPropertyName("repository_url")]
    public string? RepositoryUrl { get; set; }
}

File: 3_tier_Project_01_NodeJS_Java_MySql/backend/src/main/java/com/devopsshack/projectops/dto/ProjectRequest.java
package com.devopsshack.projectops.dto;

import com.devopsshack.projectops.entity.Priority;
import com.devopsshack.projectops.entity.ProjectStatus;
import jakarta.validation.constraints.*;
import java.time.LocalDate;

public record ProjectRequest(
        @NotBlank @Size(max = 120) String name,
        @NotBlank @Size(max = 1000) String description,
        @NotBlank @Size(max = 80) String owner,
        @NotBlank @Size(max = 80) String technology,
        @NotBlank @Size(max = 40) String environment,
        @NotNull ProjectStatus status,
        @NotNull Priority priority,
        @NotNull @Min(0) @Max(100) Integer progress,
        LocalDate dueDate
) {}
